File: web-admin/src/services/inspectionService.ts
// Inspection Service

import { apiRequest } from './api'
import type {
  Inspection,
  InspectionListResponse,
  CreateInspectionRequest,
  UpdateInspectionRequest,
  SubmitInspectionRequest,
  InspectionStatus,
} from '@/types/inspection'

export const inspectionService = {
  // List inspections with filters and pagination
  async list(params?: {
    status?: InspectionStatus
    area?: string
    siteCode?: string
    page?: number
    size?: number
  }): Promise<InspectionListResponse> {
    const queryParams = new URLSearchParams()

    if (params?.status) queryParams.append('status', params.status)
    if (params?.area) queryParams.append('area', params.area)
    if (params?.siteCode) queryParams.append('siteCode', params.siteCode)
    if (params?.page !== undefined) queryParams.append('page', params.page.toString())
    if (params?.size !== undefined) queryParams.append('size', params.size.toString())

    const queryString = queryParams.toString()
    const url = `/inspections${queryString ? `?${queryString}` : ''}`

    return apiRequest<InspectionListResponse>('GET', url)
  },

  // Get single inspection by SO number
  async get(soNumber: string): Promise<Inspection> {
    return apiRequest<Inspection>('GET', `/inspections/${soNumber}`)
  },

  // Create new inspection
  async create(data: CreateInspectionRequest): Promise<{ soNumber: string; message: string }> {
    return apiRequest('POST', '/inspections', data)
  },

  // Update inspection
  async update(soNumber: string, data: UpdateInspectionRequest): Promise<{ message: string }> {
    return apiRequest('PUT', `/inspections/${soNumber}`, data)
  },

  // Submit inspection
  async submit(
    soNumber: string,
    data: SubmitInspectionRequest
  ): Promise<{ message: string; status: string }> {
    return apiRequest('POST', `/inspections/${soNumber}/submit`, data)
  },
}

File: ios-app/Services/InspectionService.swift
//
//  InspectionService.swift
//  PHA Inspection App
//
//  Inspection API Service
//

import Foundation

class InspectionService {
    static let shared = InspectionService()

    private init() {}

    // MARK: - Get Inspections List
    func getInspections(
        status: InspectionStatus? = nil,
        area: String? = nil,
        siteCode: String? = nil,
        page: Int = 0,
        size: Int = 20
    ) async throws -> InspectionListResponse {
        var endpoint = APIConfig.Endpoints.inspections + "?page=\(page)&size=\(size)"

        if let status = status {
            endpoint += "&status=\(status.rawValue)"
        }
        if let area = area {
            endpoint += "&area=\(area)"
        }
        if let siteCode = siteCode {
            endpoint += "&siteCode=\(siteCode)"
        }

        return try await APIService.shared.request(endpoint: endpoint)
    }

    // MARK: - Get Single Inspection
    func getInspection(soNumber: String) async throws -> Inspection {
        let endpoint = APIConfig.Endpoints.inspection(soNumber)
        return try await APIService.shared.request(endpoint: endpoint)
    }

    // MARK: - Create Inspection
    func createInspection(request: CreateInspectionRequest) async throws -> InspectionCreateResponse {
        return try await APIService.shared.request(
            endpoint: APIConfig.Endpoints.inspections,
            method: .post,
            body: request
        )
    }

    // MARK: - Update Inspection
    func updateInspection(soNumber: String, request: UpdateInspectionRequest) async throws {
        let endpoint = APIConfig.Endpoints.inspection(soNumber)
        let _: EmptyResponse = try await APIService.shared.request(
            endpoint: endpoint,
            method: .put,
            body: request
        )
    }

    // MARK: - Submit Inspection
    func submitInspection(soNumber: String, request: SubmitInspectionRequest) async throws {
        let endpoint = APIConfig.Endpoints.submitInspection(soNumber)
        let _: EmptyResponse = try await APIService.shared.request(
            endpoint: endpoint,
            method: .post,
            body: request
        )
    }
}

// MARK: - Empty Response Helper
private struct EmptyResponse: Codable {}
